src/utils/validateNewUser.ts

```diff
@@ -0,0 +1,15 @@
+import { expectedNewUser, testPassword } from './';
+
+export const validateNewUser = (body: any) => {
+ const { error } = expectedNewUser.validate(body);
+ if (error) {
+    return error.message;
+ }
+
+ const passwordError = testPassword(body.password);
+ if (passwordError) {
+    return passwordError;
+ }
+
+ return null;
+};
```

Comment:
still you haven't optimized this function you can call it validations let's me give sample implementation how it's should work:
``export const validate =
  (schema: typeof ObjectSchema, type: requestType)=>(req,res,_:NextFunction)=> ``
  the request type can be body, params or query,
  name those variable as you want